File: src/AdminScripts/misc/clean_bad_files.pl

$RESTDIR = "/w/halld-scifs1a/data_monitoring/RunPeriod-2014-10/ver12/REST";
$SAVEDIR = "/w/halld-scifs1a/data_monitoring/RunPeriod-2014-10/ver12/REST/badfiles";

system "mkdir -p $SAVEDIR";

@lines = `find $RESTDIR -name \*.hddm -exec ls -lh '{}' \\;`;
#@lines = split '\n',$output;

foreach $line (@lines) {
    chomp $line;
    @tokens = split / /,$line;
    #print "${tokens[9]}  ${tokens[4]}\n";

    if( $tokens[4] == "4.9K" ) {
	print "moving ${tokens[9]} ...\n";
	system "mv ${tokens[9]} /w/halld-scifs1a/data_monitoring/RunPeriod-2014-10/ver12/REST/badfiles";
    }
}

File: src/AdminScripts/misc/clean_bad_files.py
import os,sys
import subprocess

RESTDIR = "/w/halld-scifs1a/data_monitoring/RunPeriod-2014-10/ver11/REST"
SAVEDIR = "/w/halld-scifs1a/data_monitoring/RunPeriod-2014-10/ver11/REST/badfiles"


cmd= "find %s -name \*.hddm -exec ls -lh '{}' \\;"%RESTDIR
#cmd= "find %s -name \*.hddm -exec'"%RESTDIR

#p1 = subprocess.Popen(cmd.split() + ["ls -lh '{}'","\;"], stdout=subprocess.PIPE)
p1 = subprocess.Popen(cmd.split(), stdout=subprocess.PIPE)
output = p1.communicate()[0]


for line in output.splitlines():
    tokens = line.split()
    print "%s %s"%(tokens[8],tokens[4])
